117-v1-CRD-API-Conversion-Tool.md

```diff
@@ -0,0 +1,135 @@
+
+#### Workflow
+
+Requiring users to do the manual steps as part of the upgrade process is unfortunate.
+However, from experience we know that many users use different versions of Strimzi on the same Kubernetes cluster.
+So it is not easy to automate the conversion, for example, at Cluster Operator startup.
+While that would make it easier for some users, it might be a breaking change for other users.
+Therefore, we propose sticking with the manual process used for the `v1beta2` migration. 
+It may not be perfect, but it is feasible for all users.
+
+#### Conversions
+
+Most of the changes in our custom resources between `v1beta2` and `v1` are fields that are deprecated and removed.
+But some of the changes also include fields that changed, moved, etc.
+The Conversion Tool will aim to handle as many of the changes as possible.
+However, it will not be able to handle all of them (as already covered in the [Strimzi `v1` CRD API and 1.0.0 release](./113-Strimzi-v1-CRD-API-and-1.0.0-release.md) proposal).
+The conversion is not possible, for example, when it involves the handling of secrets or environment variables that must be mounted or mapped.
+
+The changes which the Conversion Tool will not be able to handle are:
+* Change from `type: oauth` authentication to `type: custom` for both Kafka brokers as well as for the client-based operands
+* Change from `type: keycloak` and `type: opa` authorization to `type: custom` authorization
+* Removal of the support for mounting Secrets as part of the `type: custom` authentication
+* Removal of the `externalConfiguration` section in the `KafkaConnect` CR
+
+However, the Conversion Tool will detect these changes, fail the conversion, and request the user to do the conversion manually before re-running the tool.
+
+### Release process
+
+The Conversion Tool will live as an additional module in the main Strimzi Operators repository (and will be removed after the 0.52 / 1.0.0 version once it is not needed anymore).
+The module name will be `v1-api-conversion` to differentiate it from the previous version called `api-conversion`.
+That will allow it to use the same `api` module version as is currently under development.
+It will also allow us to release it as part of the regular Strimzi release process.
+
+## Documentation
+
+Documentation will include the procedure for how to use the Conversion Tool to convert the resources together with the documentation of the manual conversion process as well as the upgrade to the `v1` API version in the Strimzi 0.52 / 1.0.0 version.
+The migration process is described in detail in the main [_Strimzi v1 CRD API and 1.0.0 release_ proposal](https://github.com/strimzi/proposals/blob/main/113-Strimzi-v1-CRD-API-and-1.0.0-release.md).
+
+### Testing
+
+The conversion of the Strimzi custom resources will be tested using unit and integration tests in the Conversion Tool module itself.
+Additionally, System Tests will be added to test the whole lifecycle of the conversion:
```

Comment:
I guess it's implicit that these STs will also be removed after the 0.52 / 1.0.0 version.